pkg/kubectl/describe.go

```diff
@@ -171,6 +173,65 @@ func DescriberFor(kind schema.GroupKind, c clientset.Interface) (Describer, bool
 	return f, ok
 }
 
+// GenericDescriberFor returns a generic describer for the specified mapping
+// that uses only information available from runtime.Unstructured
+func GenericDescriberFor(
+	gvk schema.GroupVersionKind,
+	discovery discovery.CachedDiscoveryInterface,
+	dynamic *dynamic.Client,
+	events coreclient.EventsGetter) Describer {
+
+	return &genericDescriber{gvk, discovery, dynamic, events}
+}
+
+type genericDescriber struct {
+	gvk       schema.GroupVersionKind
+	discovery discovery.CachedDiscoveryInterface
+	dynamic   *dynamic.Client
+	events    coreclient.EventsGetter
+}
+
+func (g *genericDescriber) Describe(namespace, name string, describerSettings DescriberSettings) (output string, err error) {
+	if !g.discovery.Fresh() {
+		g.discovery.Invalidate()
+	}
+	resources, err := g.discovery.ServerResourcesForGroupVersion(g.gvk.GroupVersion().String())
+	if err != nil {
+		return "", err
+	}
+	var apiResource *metav1.APIResource
```

Comment:
I'll take a mapping and build `metav1.APIResource` manually to satisfy `g.dynamic.Resource(apiResource, namespace)`